01-node-tutorial/answers/writeWithPromisesThen.js

```diff
@@ -0,0 +1,25 @@
+const { writeFile, readFile } = require("fs").promises;
+
+writeFile('temp.txt', 'This is my line 1')
+     .then(() => {
+    return writeFile('temp.txt','This is my line 2')
+   })
+
+   .then(() => {
+    return writeFile('temp.txt','This is my line 3')
+   })
```

Comment:
Requires changes:
The subsequent `writeFile` invocations are overwriting the previous. Instead of having 3 lines being added to `temp.txt`, they're overwriting each other! You're missing a `flag: "a"` option that appends lines to the same file when using `writeFile`. 
```js
writeFile(...).then(() => return writeFile("...", "...", { flag: "a" });
```

Lastly, less importantly, the first line shouldn't be prefixed with a new line character. Instead of `\nLine One`, it should be `Line One\n` then line two would be `Line Two\n` and then line three would just be `Line Three`.

Currently, `temp.txt` is being created with:
```
1. Line Three
```
With the fixes, it'd be:
```
1. Line One
2. Line Two
3. Line Three
```